File: Math/CPP/minimum_time_visiting_all_points.cpp
#include <cmath>
#include <vector>
using namespace std;

class Solution {
public:
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        int n = points.size(), res = 0;
        if (n == 1) return 0;
    for (int i = 0; i < points.size() - 1; i++) {
            int a = points[i][0], b = points[i][1],
                c = points[i+1][0], d = points[i+1][1],
                shortDistanceXY = min(abs(a-c), abs(b-d)),
                longDistannceXY = max(abs(a-c), abs(b-d));
            res += shortDistanceXY + (longDistannceXY - shortDistanceXY);
        }
        return res;
    }
};

File: Math/Python/minimum_time_visiting_all_points.py
class Solution:

    def helper(self, i, j):
        [a, b], [c, d] = i, j
        shortDistanceXY = min(abs(a-c), abs(b-d))
        longDistannceXY = max(abs(a-c), abs(b-d))
        return shortDistanceXY + (longDistannceXY - shortDistanceXY)

    def minTimeToVisitAllPoints(self, points: list[list[int]]) -> int:
        n = len(points)
        ans = 0
        if n == 1: return 0
        for i in range(n - 1):
            ans += self.helper(points[i], points[i+1])
        return ans
